srcs/raycasting/draw_navigator.c

```diff
@@ -0,0 +1,123 @@
+ * this value was arbitrary.
+ */
+static void	set_navigation_x_borders(t_pair_i *start, t_pair_i *end, t_cub *cub)
+{
+	int	nav_width;
+
+	nav_width = cub->pixel_per_square * cub->map_width;
+	start->x = 0;
+	end->x = cub->map_width;
+	if (nav_width > WIDTH / 3)
+	{
+		start->x = (int)(cub->pos->x - (double)WIDTH / (6.0 * (double)cub->pixel_per_square));
+		if (start->x < 0)
+			start->x = 0;
+		end->x = start->x + WIDTH / (3 * cub->pixel_per_square);
+		if (end->x > cub->map_width)
+		{
+			end->x = cub->map_width;
+			start->x = end->x - WIDTH / (3 * cub->pixel_per_square);
+		}
+	}
+}
+
+static void	set_navigation_y_borders(t_pair_i *start, t_pair_i *end, t_cub *cub)
+{
+	int	nav_height;
+
+	nav_height = cub->pixel_per_square * cub->map_height;
+	start->y = 0;
+	end->y = cub->map_height;
+	if (nav_height > HEIGHT / 3)
+	{
+		start->y = (int)(cub->pos->y - (double)HEIGHT / (6.0 * (double)cub->pixel_per_square));
+		if (start->y < 0)
+			start->y = 0;
+		end->y = start->y + HEIGHT / (3 * cub->pixel_per_square);
+		if (end->y > cub->map_height)
+		{
+			end->y = cub->map_height;
+			start->y = end->y - HEIGHT / (3 * cub->pixel_per_square);
+		}
+	}
+}
+
+/* The counters i and j represent the point in the map grid
+ * we are looking at. To convert it to the pixel screen,
+ * one must remove the offset and * cub->pixel_per_square
+*/
+static void	actual_draw_navigator(t_cub *cub, t_pair_i start, t_pair_i end)
+{
+	int	i;
+	int	j;
+
+	i = start.y - 1;
+	while (i < end.y + 1)
+	{
+		j = start.x - 1;
+		while (j < end.x + 1)
+		{
+			if (i == start.y - 1 || i == end.y || j == start.x - 1 || j == end.x)
+				draw_square(cub, (j - start.x + 1) * cub->pixel_per_square,
+					(i - start.y + 1) * cub->pixel_per_square, 0x00000000);
+			else if (cub->map[i][j] == OUT)
+				draw_square(cub, (j - start.x + 1) * cub->pixel_per_square,
+					(i - start.y + 1) * cub->pixel_per_square, 0x00000000);
+			else if (cub->map[i][j] == IN || cub->map[i][j] == PLAYER)
+				draw_square(cub, (j - start.x + 1) * cub->pixel_per_square,
+					(i - start.y + 1) * cub->pixel_per_square, 0x00333333);
+			else if (cub->map[i][j] == WALL)
+				draw_square(cub, (j - start.x + 1) * cub->pixel_per_square,
+					(i - start.y + 1) * cub->pixel_per_square, 0x00FFFFFF);
+			j++;
+		}
+		i++;
+	}
+}
+
+/* Draw Navigator 🗺
+ * - sets the borders of our little navigator: if the original map
+ *   is too big, then only a part is shown
+ * - draws the navigator
+ * - draws the player as a red square: tried to add a small line
+ *   to show in which direction the player was looking, but didn't
+ *   look very well...
+ */
+void	draw_navigator(t_cub *cub)
+{
+	t_pair_i	start;
+	t_pair_i	end;
+
+	set_navigation_x_borders(&start, &end, cub);
+	set_navigation_y_borders(&start, &end, cub);
+	actual_draw_navigator(cub, start, end);
+	draw_square(cub, ((int)cub->pos->x - start.x + 1) * cub->pixel_per_square,
+		((int)cub->pos->y - start.y + 1) * cub->pixel_per_square, 0x00FF0000);
```

Comment:
This is what draws the red square where we are right?

As the map is for navigation, I was wondering if it would perhaps be possible to make the player location in a sort of triangle that shows the direction we are currently facing? I got confused a bit sometimes with the minimap because this isn't indicated on it.

Let me know what you think!

Reply:
The last draw_square in line 121 is what draws the player yes.
I thought about it... I also thought about doing a small line in front of the player representing the direction it was looking at. I must say it didn't look well 😅
I can think about it, but it will take a bit as I don't actually know how to make a triangle, I need to look into that

Reply:
Another comment is that the grid system can be a bit "jumpy". Fine for me to stick with it (it's bonus anyway so not sure if we should spend this much time on it), but if it's not hard smoother movement (redrawing a player 1px to the right or left) without the forced grid structure could be nice.

Again, not really priority, as we should probably focus most on textures which I am still quite lost on right now 😆